### ISSUE TITLE
Creating private URL not possible for the role "Contributor"

### ISSUE THREAD

**Issue by sergejzr** [IQSS/dataverse#9938]
Dear Dataverse Community,
we were looking for ways the standard roles (e.g. "Curator", "Contributor", etc.) can be edited, but could not find a solution so far. 

The concrete problem we are facing at the moment is that a Contributor can not create private URLs after uploading a dataset. This always needs to be done by administrators at the moment. I guess it is because "Contributor" is missing the permisson "ManageDatasetPermissions".

Creating a new Role with required permissions does not solve the problem, because it can not be automatically assigned to a dataset uploader (unlike the "Contributor" role)

There might be other ways to achieve the behavior, but as I mentioned, we could not find any, unfortunately.

Thank you for your thoughts on this!
Sergej

**Comment by pdurbin** [comment]
Right, CreatePrivateUrlCommand calls AssignRoleCommand which requires ManageDatasetPermissions (for a dataset).

It look like the user would have enough permission as a Curator (at the dataset level) to create a Private URL...

![Screen Shot 2023-09-20 at 11 27 57 AM](https://github.com/IQSS/dataverse/assets/21006/bca0d898-680e-4870-ae41-a3878a53b678)

... but is that too much power?

**Comment by sergejzr** [comment]
Thanks Phil, yes, we have to run through an internal review process before publishing the dataset, so the "PublishDataset" permission would be too much, also adding Dataverses should not be possible for any user who just published a dataset...

**Comment by DS-INRA** [comment]
Hello,
you can actually set a custom role as the  default role for dataset creators, but it has to be done via API (https://guides.dataverse.org/en/latest/api/native-api.html#assign-default-role-to-user-creating-a-dataset-in-a-dataverse-collection).
I guess the issue is then twofold :
- ability to add other roles as default via the UI (I don't think there's any issue on this)
- distinguish the permission to create private and the managedatasetpermissions permission (which gives you the right to set yourself or others as admins and therefore to bypass the restriction on publication), I was certain there was an issue on this but I didn't find it with my quick search

**Comment by sergejzr** [comment]
Absolutely right, thanks!
The temporary solution to set a default role using the API works and we can use it for now, i think, although the additional access permissions set by users my caus some troubles.

I aslo absolutely agree with your splitting the problem into two those subproblems.

**Comment by pdurbin** [comment]
There's a related discussion kicked off by @vera here: https://groups.google.com/g/dataverse-community/c/wZfSTBiJuPQ/m/F_feNKA1BwAJ

**Comment by pdurbin** [comment]
Also related:

- #9358

**Comment by DS-INRA** [comment]
I went ahead and created this issue for the UI part, I didn't add one to split off a ManagePrivateUrl permission, as I'm currently not sure that wether or not https://github.com/IQSS/dataverse/issues/9358  would be best at least for us, @sergejzr feel free to create it or rename this issue if necessary for you :)

**Comment by DS-INRA** [comment]
Hello again,
I finally decided to create an issue for a ManagePrivateUrl permission, to have a more detailed issue on this specific aspect:
- https://github.com/IQSS/dataverse/issues/10574

**Comment by cmbz** [comment]
2025-03-24
- Closing because this issue now covers the specific use case needed to implement the feature: https://github.com/IQSS/dataverse/issues/10574

**Comment by pdurbin** [comment]
I just reopened this issue because it describes the problem quite well, a problem that @Falco-KUB recently brought up again at https://groups.google.com/g/dataverse-community/c/ZZpnOTuWOVQ/m/27emDwNRBgAJ

A possible solution is to have a dedicated permission for creating/deleting Preview URLs:

- #10574

There's also this new issue that's highly related:

- https://github.com/IQSS/dataverse-pm/issues/599

**Comment by sbarbosadataverse** [comment]
> I just reopened this issue because it describes the problem quite well, a problem that [@Falco-KUB](https://github.com/Falco-KUB) recently brought up again at https://groups.google.com/g/dataverse-community/c/ZZpnOTuWOVQ/m/27emDwNRBgAJ
> 
> A possible solution is to have a dedicated permission for creating/deleting Preview URLs:
> 
> * [Add a ManagePrivateURL permission #10574](https://github.com/IQSS/dataverse/issues/10574)
> 
> There's also this new issue that's highly related:
> 
> * [Feature Request/Idea: Clarify interaction between Private URLs and “Submit for Review” to prevent unintended publication dataverse-pm#599](https://github.com/IQSS/dataverse-pm/issues/599)

@pdurbin the new role we created in HDV poroduction allows private URL feature + general contributor permissions. I am sure you referenced this somewhere in a related ticket

**Comment by pdurbin** [comment]
@sbarbosadataverse yes, I didn't want to actively promote that custom role in my post to the mailing list (too complicated) but I'll paste here some technical details from [Slack](https://iqss.slack.com/archives/GRDNRJ6BZ/p1785438794881689?thread_ts=1785178956.983949&cid=GRDNRJ6BZ):

<img width="788" height="1136" alt="Image" src="https://github.com/user-attachments/assets/27723707-d222-4d92-8f64-c08b3b070b0c" />
